models/run.py

```diff
@@ -48,96 +140,130 @@
-                human.timestamp = m_i[0]
-                # update risk based on that day's messages
-                RiskModel.update_risk_encounter(human, m_i)
-
-            # go about your day and accrue encounters
-            encounters = logs[hash_id_day(human.name, current_day)]
-            for encounter in encounters:
-                # extract variables from log
-                encounter_time = encounter['time']
-                unobs = encounter['payload']['unobserved']
-                encountered_human = hd[unobs['human2']['human_id']]
-                human.messages.append(encountered_human.cur_message(encounter_time))
-
-
-            if start_risk > human.risk + 0.1 or start_risk < human.risk - 0.1:
-                for m in human.messages:
-                    # if the encounter happened within the last 14 days, and your symptoms started at most 3 days after your contact
-                    if todays_date - m.time < datetime.timedelta(days=14) and human.symptoms_start < m.time + datetime.timedelta(days=3):
-                        hd[m.unobs_id].messages.append(human.cur_message(encounter_time))
-
-            # append the updated risk for this person and whether or not they are actually infectious
-            daily_risks.append((human.risk, human.is_infectious, human.name))
-        if PLOT_DAILY:
-            hist_plot(daily_risks, f"{PATH_TO_PLOT}day_{str(current_day).zfill(3)}.png")
-        all_risks.extend(daily_risks)
+    for current_day in tqdm(range(days)):
+        start1 = time.time()
         daily_risks = []
-    dist_plot(all_risks,  f"{PATH_TO_PLOT}all_risks.png")
+
+        with Parallel(n_jobs=args.n_jobs, batch_size='auto', verbose=10) as parallel:
+            all_params = []
+            for human in hd.values():
+                encounters = logs[hash_id_day(human.name, current_day)]
+                all_params.append({"start": start, "current_day": current_day, "RiskModel": RiskModel, "encounters": encounters, "rng": rng, "all_possible_symptoms": all_possible_symptoms, "human": human})
+                # go about your day accruing encounters and clustering them
+                for idx, encounter in enumerate(encounters):
+                    encounter_time = encounter['time']
+                    unobs = encounter['payload']['unobserved']
+                    encountered_human = hd[unobs['human2']['human_id']]
+                    message = encountered_human.cur_message(current_day, RiskModel)
+                    encountered_human.sent_messages[
+                        str(unobs['human1']['human_id']) + "_" + str(encounter_time)] = message
+                    human.messages.append(message)
+                    got_exposed = encounter['payload']['unobserved']['human1']['got_exposed']
+                    if got_exposed:
+                        human.exposure_message = _encode_message(message)
+
+            daily_output = parallel((delayed(proc_human)(params) for params in all_params))
```

Comment:
`daily_output = parallel((delayed(proc_human)(args, params) for params in all_params))`

Reply:
not necessary with the change I made